src/main/java/orca/flukes/GUIUnifiedState.java

```diff
@@ -877,6 +878,39 @@ public void run() {
 					// query one last time to get e.g. IP addresses and other labels with late binding
 					queryManifest();
 				}
+			} else if (e.getActionCommand().equals("addsshkey")) {
+			    try {
+                    // display key dialog
+                    NewUserDialog sshKeyDialog = new NewUserDialog(GUI.getInstance().getFrame(), "Account and SSH keys", 
+                            "Paste your public SSH key here:", 20, 50);
+                    sshKeyDialog.pack();
+                    sshKeyDialog.setVisible(true);
+                    String keys = sshKeyDialog.getSSHKeys();
+                    boolean sudo = sshKeyDialog.getSudo();
+                    String username = sshKeyDialog.getUsername();
+
+                    for(OrcaNode on: g.getVertices()) {
+                        if (on instanceof OrcaNode) {
```

Comment:
IntelliJ points out
>Condition 'on instanceof OrcaNode' is redundant and can be replaced with '!= null'

Reply:
It should probably say

for (OrcaResource on: g.getVertices()) {
    if (on instanceof OrcaNode) {
   }
}